launcher/ui/pages/instance/McClient.cpp

```diff
@@ -0,0 +1,165 @@
+#include <QObject>
+#include <QTcpSocket>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include <Exception.h>
+#include "McClient.h"
+#include <qtconcurrentrun.h>
+
+#define SEGMENT_BITS 0x7F
+#define CONTINUE_BIT 0x80
+
+McClient::McClient(QObject *parent, QString domain, QString ip, short port): QObject(parent), domain(domain), ip(ip), port(port) {}
+
+QJsonObject McClient::getStatusDataBlocking() {
+    qDebug() << "Connecting to socket..";
+    socket.connectToHost(ip, port);
+
+    if (!socket.waitForConnected(3000)) {
+        throw Exception("Failed to connect to socket");
+    }
+    qDebug() << "Connected to socket successfully";
+    sendRequest();
+
+    if (!socket.waitForReadyRead(3000)) {
+        throw Exception("Socket didn't send anything to read");
+    }
+    return readResponse();
+}
+
+QFuture<int> McClient::getOnlinePlayers() {
+    return QtConcurrent::run([this]() {
+        try {
+            auto status = getStatusDataBlocking();
+            int onlinePlayers = status.value("players").toObject().value("online").toInt();
```

Comment:
You can adapt this to use the https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/Json.h functions like:`requireInt`

Reply:
What is the advantage of this, other than being quicker to write ?